cpp/arcticdb/vector_db/vector_db.hpp

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#pragma once
+
+#include <arcticdb/processing/clause.hpp>
+
+namespace arcticdb {
+
+    struct NamedColumnSortedByDistance {
+        double distance_;
+        std::shared_ptr<Column> column_;
+        std::string_view name_;
+
+        explicit NamedColumnSortedByDistance(double distance,
+                                             std::shared_ptr<Column> column,
+                                             std::string_view name) :
+                distance_(distance),
+                column_(column),
+                name_(name) {}
+
+        NamedColumnSortedByDistance() = delete;
+
+        ARCTICDB_MOVE_COPY_DEFAULT(NamedColumnSortedByDistance);
+
+        bool operator<(const NamedColumnSortedByDistance &that) const {
+            return distance_ < that.distance_ || (distance_ == that.distance_ && name_ < that.name_);
+            // name is a tiebreaker.
+        };
+
+        bool operator==(const NamedColumnSortedByDistance &that) const {
+            return distance_ == that.distance_ && column_ == that.column_ && name_ == that.name_;
+        };
+
+    };
+
+    struct TopK {
+        uint64_t _k;
+        std::set<NamedColumnSortedByDistance> top_k_;
```

Comment:
nit-picking, but underscores after the variable in C++

Reply:
Hm I think I had some notion of trying to indicate it'd be private and forgot I could simply _make_ it private were I bothered.